trieste/space.py

```diff
@@ -329,3 +346,124 @@ def __mul__(self, other: Box) -> Box:
 
     def __deepcopy__(self, memo: dict[int, object]) -> Box:
         return self
+
+
+class ProductSearchSpace(SearchSpace):
```

Comment:
Maybe make explicit the fact that the subspaces are tagged? (which differentiates this e.g. from the more limited multiplication method on spaces)

```suggestion
class TaggedProductSearchSpace(SearchSpace):
```

Reply:
Yep I like this idea

Reply:
actually, now that we can have default tagging behaviour, I will keep it as ProductSearchSpace